lib/assets/assets.go

```diff
@@ -53,14 +54,18 @@ func Serve(w http.ResponseWriter, r *http.Request, asset Asset) {
 		return
 	}
 
-	if strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+	header.Set("Content-Length", strconv.Itoa(len(asset.Content)))
+
+	switch {
+	case !asset.Gzipped:
+		io.WriteString(w, asset.Content)
+	case strings.Contains(r.Header.Get("Accept-Encoding"), "gzip"):
 		header.Set("Content-Encoding", "gzip")
-		header.Set("Content-Length", strconv.Itoa(len(asset.ContentGz)))
-		io.WriteString(w, asset.ContentGz)
-	} else {
+		io.WriteString(w, asset.Content)
+	default:
 		// gunzip for browsers that don't want gzip.
 		var gr *gzip.Reader
-		gr, _ = gzip.NewReader(strings.NewReader(asset.ContentGz))
+		gr, _ = gzip.NewReader(strings.NewReader(asset.Content))
```

Comment:
wrong content-length here?

Reply:
Yes. Should be fixed now, and I added an assertion to the test.